ArborGVT/src/BarnesHutTree.cs

```diff
@@ -186,9 +186,9 @@ public void applyForces(ArborNode m, double g)
                         ptx = branch.Pt.div(massx);
 
                         k = m.Pt.sub(ptx);
-                        kMag = k.magnitude();
+                        kMag = k.magnitudeSquare();
 
-                        double h = Math.Sqrt(branch.Size.X * branch.Size.Y);
+                        double h = branch.Size.X * branch.Size.Y;
                         if (h / kMag > fDist)

```

Comment:
Remember `fDist` is a square now? `h / kMag` also is an expression from under `SQRT`.